abci/client/creators.go

```diff
@@ -1,11 +1,8 @@
-package proxy
+package abcicli
```

Comment:
If we're going to move this package anyway, I'd like to advocate we at least not abbreviate it (or maybe even find a better name), and also take the opportunity to reduce some of the stutter that already exists in the package (this should I think be easy, since we're making it internal).

Some suggestions for your delectation:

```go
package abciclient

type Creator interface {
   NewClient() (Client, error)
}

func NewDefaultCreator(addr, transport, db string) (Creator, io.Closer) { /* … */ }

func NewLocalCreator(addr, transport, db string) (Creator, io.Closer) { /* … */ }
func NewLocalClient() (Client, error) { /* … */ }
// etc.
```